traffic_ops/traffic_ops_golang/role/roles.go

```diff
@@ -159,16 +159,32 @@ func (role *TORole) deleteRoleCapabilityAssociations(tx *sqlx.Tx) (error, error,
 }
 
 func (role *TORole) Read() ([]interface{}, error, error, int) {
+	version := role.APIInfo().Version
+	if version == nil {
+		return nil, nil, errors.New("TORole.Read called with nil API version"), http.StatusInternalServerError
+	}
+	if version.Major != 1 || version.Minor < 1 {
+		return nil, nil, fmt.Errorf("TORole.Read called with invalid API version: %d.%d", version.Major, version.Minor), http.StatusInternalServerError
+	}
```

Comment:
I don't think you need to do this, because that check ought to be caught by [some middleware](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/routing/wrappers.go#L281-L287).

Reply:
Yeah, I was on the fence about whether or not to include this check. This condition would only be true if there was a bug introduced into the routing middleware, so really it's just an extra safety check in case that ever happens. I can remove it.